packages/gafl-webapp-service/src/pages/order-complete/licence-details/__tests__/route.spec.js

```diff
@@ -26,149 +26,156 @@ jest.mock('../../../../constants.js', () => ({
-        await expect(getData(mockRequest)).rejects.toThrow(boomError)
-      })
+const getMockStatus = () => ({
+  [COMPLETION_STATUS.agreed]: 'agreed',
+  [COMPLETION_STATUS.posted]: 'posted',
+  [COMPLETION_STATUS.finalised]: 'finalised',
+  [CommonResults.OK]: 'ok',
+  [ShowDigitalLicencePages.YES]: 'yes'
+})
 
-      it('if status finalised flag is not set', async () => {
-        const mockRequest = createMockRequest({ status: { [COMPLETION_STATUS.agreed]: true, [COMPLETION_STATUS.posted]: true } })
-        const boomError = Boom.forbidden('Attempt to access the licence information handler with no finalised flag set')
-        await expect(getData(mockRequest)).rejects.toThrow(boomError)
-      })
-    })
+const getMockTransaction = () => ({
+  permissions: [getMockPermission()]
+})
 
-    it('calls licenceTypeDisplay with permission and i18n catalog', async () => {
-      const i18nCatalog = Symbol('mock catalog')
-      const permission = Symbol('mock permission')
+const getMockPermission = () => ({
+  licensee: {
+    firstName: 'Turanga',
+    lastName: 'Leela',
+    obfuscatedDob: 'obfuscatedDob'
+  },
+  referenceNumber: '123456789',
+  licenceType: 'trout-and-coarse',
+  numberOfRods: '2',
+  licenceLength: '8D',
+  permit: { cost: 12 }
+})
 
-      const sampleRequest = createMockRequest({ permission, i18nCatalog })
+const getMockCatalog = overrides => ({
+  age_senior_concession: Symbol('Over 65'),
+  age_junior_concession: Symbol('Junior (13 to 16)'),
 
-      await getData(sampleRequest)
+  licence_type_1d: Symbol('1 day'),
+  licence_type_8d: Symbol('8 days'),
+  licence_type_12m: Symbol('12 months'),
+  ...overrides
+})
 
-      expect(licenceTypeDisplay).toHaveBeenCalledWith(permission, i18nCatalog)
+describe('licence-length > route', () => {
+  beforeEach(() => jest.clearAllMocks())
+
+  describe('getData', () => {
+    const getMockRequest = (statusGet = () => {}, transactionGet = () => {}, catalog = getMockCatalog()) => ({
+      cache: () => ({
+        helpers: {
+          transaction: {
+            get: async () => transactionGet
+          },
+          status: {
+            get: async () => statusGet
+          }
+        }
+      }),
+      i18n: {
+        getCatalog: () => catalog
+      }
     })
 
-    it('return value of licenceTypeDisplay is used for licenceTypeStr', async () => {
-      const licenceDisplay = Symbol('licence display')
-      licenceTypeDisplay.mockReturnValueOnce(licenceDisplay)
+    const mockRequest = getMockRequest(getMockStatus(), getMockTransaction())
+    const mssgs = mockRequest.i18n.getCatalog()
 
-      const { licenceTypeStr } = await getData(createMockRequest())
+    beforeEach(() => jest.clearAllMocks())
 
-      expect(licenceTypeStr).toBe(licenceDisplay)
+    it('licenceTypeDisplay is called with permissions and mssgs', async () => {
+      await getData(mockRequest)
+      expect(licenceTypeDisplay).toHaveBeenCalledWith(getMockPermission(), mssgs)
```

Comment:
You're getting a second instance of an object and comparing it with the first. Don't do this, create the permission, pass it to the mock request generator and then check that licenceTypeDisplay has been called with the same object.